utils/launcherUtils.py

```diff
@@ -439,25 +576,116 @@ def launch(URL, MOD_ID, MOD_NAME, LINK_TYPE):
+            while not (exists(InstallDir + "/data/iso_data/" + GAME + "//Z6TAIL.DUP")):
+                print(InstallDir + "/data/iso_data/" + GAME + "//Z6TAIL.DUP " + "Not found yet, sleeping then checking again to make sure the install finished")
+                time.sleep(5)
+
+            #open GoalC to build jak2, for jak 1 extractor can handle this.
+            print("Opening the Compiler subprocess - Sleeping for 5 seconds so it has time to initalize.")
+            subprocess.Popen(goalc_command_list, shell=True, cwd=os.path.abspath(InstallDir))
+            
+
+        time.sleep(5)
+        while process_exists("extractor.exe"):
+            print("extractor.exe still running, sleeping for 1s")
+            time.sleep(1)
+
+        #jak2hack this is only needed since extractor isnt aware of jak2
+        while process_exists("goalc.exe"):
+            print("goalc.exe still running, sleeping for 1s")
+            time.sleep(1)
+
+        while not (exists(InstallDir + "/data/iso_data/" + GAME + "//Z6TAIL.DUP")) and (exists(InstallDir + "/data/iso_data/" + GAME + "//")):
+            time.sleep(5)
+
+        if (exists(InstallDir + "/data/iso_data/" + GAME + "//Z6TAIL.DUP")) and not (exists(UniversalIsoPath + "//" + GAME + "//Z6TAIL.DUP")):
+            shutil.copytree( InstallDir + "/data/iso_data/" + GAME, UniversalIsoPath + "//" + GAME + "//")
+        path_to_remove = InstallDir + "/data/iso_data/" + GAME
+        if os.path.exists(path_to_remove):
+            shutil.rmtree(path_to_remove)
+            print(f"Path '{path_to_remove}' removed successfully.")
+        else:
+            print(f"Path '{path_to_remove}' does not exist.")
+        #print("copying " + InstallDir + "/data/iso_data/" + GAME + "to " + UniversalIsoPath )
+
+
+        # at this point we should be good to launch, these are just sanity checks that should never be reached
+
+        #update the timestamp of the local exe
+        if exists(InstallDir + "/" + ExecutableName):
+            LastWrite = datetime.utcfromtimestamp(
+            pathlib.Path(InstallDir + "/" + ExecutableName).stat().st_mtime
+        )
+        NotExtracted = bool(not (exists(UniversalIsoPath + "//" + GAME + "//" + "Z6TAIL.DUP")))
+        NotCompiled = bool(not (exists(InstallDir + r"\data\out" + "//" + GAME + "//" + "fr3\GAME.fr3")))
+        needUpdate = bool((LastWrite < LatestRel) or (NotExtracted) or NotCompiled)
+        if(NotExtracted):
+            print("Error! Iso data does not appear to be extracted to " + UniversalIsoPath +"//" + GAME + "//" + "Z6TAIL.DUP")
+            print("Will ask user to provide ISO")
+        if(NotCompiled):
+            print("Error! The game is not compiled")
+        if((LastWrite < LatestRel)):
+            print("Looks like we need to download a new update!")
+            print(LastWrite)
+            print(LatestRel)
+            print("Is newest posted update older than what we have installed? " + str((LastWrite < LatestRel)))
```

Comment:
should it error out here? and not launch

Reply:
This should never get hit, but I figured checking a second time didn't hurt